### ISSUE TITLE
Rule Request: [Multiline Function Chains]

### ISSUE THREAD

**Issue by erichoracek** [realm/SwiftLint#2214]
#### Summary

Chained function calls should be either on the same line, or one per line.

This is similar to the existing "Multiline Arguments" and "Multiline Parameters" rules.

#### Q&A

> 1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

It's easier to follow control flow through long function chains when each call has the same indentation.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

* Triggering:
```swift
return match(pattern: pattern, with: [.comment]).flatMap { range in
   return Command(string: contents, range: range)
}.flatMap { command in
   return command.expand()
}

let evenSquaresSum = [20, 17, 35, 4]
    .filter { $0 % 2 == 0 }.map { $0 * $0 }
    .reduce(0, +)
```

* Non-triggering
```swift
return match(pattern: pattern, with: [.comment])
    .flatMap { range in
        return Command(string: contents, range: range)
    }
    .flatMap { command in
        return command.expand()
    }

let evenSquaresSum = [20, 17, 35, 4].filter { $0 % 2 == 0 }.map { $0 * $0 }.reduce(0, +)

let evenSquaresSum = [20, 17, 35, 4]
    .filter { $0 % 2 == 0 }
    .map { $0 * $0 }
    .reduce(0, +)
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

No configuration necessary

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

Disabled by default

**Comment by freak4pc** [comment]
@erichoracek I like this and it's also useful to me and many friends in the Realm (😉) of Rx. 
Do you want to tackle this or are you OK with me giving it a go?

**Comment by erichoracek** [comment]
@freak4pc Thanks for the interest! I'm pretty close to having a working implementation—feel free to take a shot at this if you'd like too though 😄

**Comment by freak4pc** [comment]
@erichoracek oh not at all! Just wanted
To help out :) feel free to take your time with this

**Comment by markst** [comment]
#490
